src/layman/publication_relation/util.py

```diff
@@ -7,3 +7,28 @@ def update_related_publications_after_change(workspace, publication_type, public
         maps = get_publication_info(workspace, publication_type, publication, context={'keys': ['used_in_maps']})['used_in_maps']
         for map_obj in maps:
             patch_after_feature_change(map_obj['workspace'], MAP_TYPE, map_obj['name'])
+
+
+def check_no_internal_workspace_name_layer(map_json, *, x_forwarded_items):
+    from layman import names, LaymanError
+    from layman.map.util import get_internal_gs_layers_from_json
+    from layman.layer.prime_db_schema.table import get_layer_info
+    found_gs_layer = get_internal_gs_layers_from_json(map_json, x_forwarded_items=x_forwarded_items)
+    found_layers = []
+    for layer_idx, gs_workspace, gs_layer in found_gs_layer:
+        layer_uuid = names.geoserver_layername_to_uuid(geoserver_workspace=gs_workspace,
+                                                       geoserver_name=gs_layer)
+        if not layer_uuid:
+            workspace = gs_workspace[:-4] if gs_workspace.endswith('_wms') else gs_workspace
+            layer = get_layer_info(workspace=workspace, layername=gs_layer)
+            if layer:
```

Comment:
By the way, this is another production case where I would prefer calling `Layer(layer_tuple=(workspace, gs_layer))` to `get_layer_info(...)`, but we can't do it simply, because `Layer` class is not prepared for nonexistent layers.